service/header/store.go

```diff
@@ -17,14 +17,39 @@ var (
 	DefaultStoreCacheSize = 1024
 	// DefaultIndexCacheSize defines the amount of max entries allowed in the Height to Hash index cache.
 	DefaultIndexCacheSize = 256
+	// DefaultWriteBatchSize defines the size of the batched header write.
+	// Headers are written in batches not to thrash the underlying Datastore with writes.
+	DefaultWriteBatchSize = 2056
 )
 
+// errStoppedStore is returned for attempted operations on a stopped store
+var errStoppedStore = errors.New("header: stopped store")
+
+// store implements the Store interface for ExtendedHeaders over Datastore.
 type store struct {
-	ds    datastore.Batching
+	// header storing
+	//
+	// underlying KV store
+	ds datastore.Batching
+	// adaptive replacement cache of headers
 	cache *lru.ARCCache
-	index *heightIndexer
 
+	// header heights management
+	//
+	// maps heights to hashes
+	heightIndex *heightIndexer
+	// manages current store head height (1) and
+	// allows callers to wait till header for a height is stored (2)
 	heightSub *heightSub
+
+	// writing to datastore
+	//
+	// queued of header ranges to be written
+	writes chan []*ExtendedHeader
+	// signals when writes are finished
+	writesDn chan struct{}
```

Comment:
DN 🤣 

Does this stand for done in this case?

Reply:
Yes, previously there was an explicit done word in the commit, but I applied Rene's suggestion which changed it to finished.